Update README with how to setup and run flutter instructions.

## Description
As a developer,,
I need to make sure contributors can setup and run flutter on their local machine so they can start contributing to the project,

## Mocks
N/A

## Acceptance Criteria

- [ ] Update README.md with setup instructions

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
1 hour

Thanks, @isabelcosta and @bartekpacia for assigning me to the issue. I'm excited to learn more on this. actually I have my 1st question, @bartekpacia and @Techno-Disaster.

When I installed the flutter on my Android and when tried to register, came across same issue as [issue#32](https://github.com/anitab-org/mentorship-flutter/issues/32) so I decided to clone and run from IDE. 

I cloned and tried to run on VSCode IDE with dart and flutter extension installed but still run into an issue. Below is the console log of the error. Can you please advise what might cause this @bartekpacia and @Techno-Disaster?

```quote
Running "flutter pub get" in mentorship-flutter...                 85.5s
Launching lib/main.dart on Android SDK built for x86 in debug mode...

Compiler message:
lib/screens/home/pages/stats/stats_page.dart:47:64: Error: The getter 'headline6' isn't defined for the class 'TextTheme'.
 - 'TextTheme' is from 'package:flutter/src/material/text_theme.dart' ('../../Downloads/flutter/packages/flutter/lib/src/material/text_theme.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'headline6'.
                            style: Theme.of(context).textTheme.headline6,
                                                               ^^^^^^^^^
lib/screens/home/pages/stats/stats_page.dart:92:48: Error: The getter 'headline6' isn't defined for the class 'TextTheme'.
 - 'TextTheme' is from 'package:flutter/src/material/text_theme.dart' ('../../Downloads/flutter/packages/flutter/lib/src/material/text_theme.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'headline6'.
            style: Theme.of(context).textTheme.headline6.apply(color: Colors.grey[600]),
                                               ^^^^^^^^^
lib/screens/home/pages/stats/stats_page.dart:96:48: Error: The getter 'subtitle2' isn't defined for the class 'TextTheme'.
 - 'TextTheme' is from 'package:flutter/src/material/text_theme.dart' ('../../Downloads/flutter/packages/flutter/lib/src/material/text_theme.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'subtitle2'.
            style: Theme.of(context).textTheme.subtitle2.copyWith(fontWeight: FontWeight.bold),
                                               ^^^^^^^^^
lib/widgets/bold_text.dart:15:44: Error: The getter 'bodyText2' isn't defined for the class 'TextTheme'.
 - 'TextTheme' is from 'package:flutter/src/material/text_theme.dart' ('../../Downloads/flutter/packages/flutter/lib/src/material/text_theme.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'bodyText2'.
        style: Theme.of(context).textTheme.bodyText2,
                                           ^^^^^^^^^
Target kernel_snapshot failed: Exception: Errors during snapshot creation: null
build failed.

FAILURE: Build failed with an exception.

* Where:
Script '/Users/mayatreacy/Downloads/flutter/packages/flutter_tools/gradle/flutter.gradle' line: 780

* What went wrong:
Execution failed for task ':app:compileFlutterBuildDebug'.
> Process 'command '/Users/mayatreacy/Downloads/flutter/bin/flutter'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 5m 35s
Gradle task assembleDebug failed with exit code 1
Exited (sigterm)
```

--- a/.gitignore
+++ b/.gitignore
@@ -18,7 +18,7 @@
 # The .vscode folder contains launch configuration and tasks you configure in
 # VS Code which you may wish to be included in version control, so this line
 # is commented out by default.
-#.vscode/
+.vscode/
 
 # Flutter/Dart/Pub related
 **/doc/api/

--- a/README.md
+++ b/README.md
@@ -48,6 +48,27 @@ Occasional UI bugs may occur – in this case, please create an issue.
 - To save JWT token, [Flutter Secure Storage](https://pub.dev/packages/flutter_secure_storage) is used
 - If you find something in code that looks a bit odd, it might be some useful extension method from `lib/extensions`
 
+#### Setup instructions
+
+1. Install Flutter by following instructions from [flutter.dev](https://flutter.dev).
+2. Make sure you install the **Flutter** and **Dart** plugins. The installation instructions above should have instructions on this too.
+3. Fork and clone the [mentorship-flutter](https://github.com/anitab-org/mentorship-flutter) repository to your local machine.
+4. Open the project on your IDE. Don't be shocked if you see error warnings.
+5. Open the terminal and run `flutter channel beta` to switch from stable channel to beta channel of flutter. The beta channel is the one we are using for this project.
+6. Once done, run the project by running `flutter run` in the terminal to run the app in debug mode. To build a release build you can do the following:
+
+  *for Android app*
+
+  In the terminal, run the `flutter build apk` command. To build the apk specific to your device arch you can run `flutter build apk --split-per-abi` or `flutter build appbundle --target-platform android-arm,android-arm64,android-x64` (remove arguments which are not required) to get only your arch build. You can read more about this [here](https://flutter.dev/docs/deployment/android)
+
+  *for IOS app*
+
+  To build a release for IOS app, run `flutter build ios` from the terminal. To learn more on creating build archive, release app on TestFlight or to App Store, click [here](https://flutter.dev/docs/deployment/ios).
+
+
+
+By default the backend of this project is set to the mentorship-backend dev server. You are now set to go.
+
 ### Web support
 
 Flutter for Web is currently at technical preview stage. To test this functionality, I created

--- a/android/.gitignore
+++ b/android/.gitignore
@@ -5,3 +5,8 @@ gradle-wrapper.jar
 /gradlew.bat
 /local.properties
 GeneratedPluginRegistrant.java
+/.project
+/.settings/org.eclipse.buildship.core.prefs
+/app/.classpath
+/app/.project
+/app/.settings/org.eclipse.buildship.core.prefs

